crates/bevy_render/src/texture/image.rs

```diff
@@ -710,6 +1069,15 @@ pub enum TranscodeFormat {
     Rgb8,
 }
 
+/// An error that occurs when accessing specific pixels in a texture
+#[derive(Error, Debug)]
+pub enum TextureAccessError {
+    #[error("out of bounds ({0})")]
+    OutOfBounds(String),
+    #[error("unsupported texture format: {0}")]
+    UnsupportedTextureFormat(String),
```

Comment:
Same here, why not format later?

```suggestion
    #[error("unsupported texture format: {0:?}")]
    UnsupportedTextureFormat(TextureFormat),
```